hudi-client/hudi-client-common/src/main/java/org/apache/hudi/table/marker/DirectWriteMarkers.java

```diff
@@ -119,7 +139,7 @@ public Set<String> createdAndMergedDataPaths(HoodieEngineContext context, int pa
         while (itr.hasNext()) {
           FileStatus status = itr.next();
           String pathStr = status.getPath().toString();
-          if (pathStr.contains(HoodieTableMetaClient.MARKER_EXTN) && !pathStr.endsWith(IOType.APPEND.name())) {
+          if (pathStr.contains(HoodieTableMetaClient.INPROGRESS_MARKER_EXTN) && !pathStr.endsWith(IOType.APPEND.name())) {
             result.add(translateMarkerToDataPath(pathStr));
```

Comment:
actually we are interested only in inProgress here. 
crux here is that. 
if spark retries and in 2nd attempt, a data file was created, but never completed (possible in hdfs), and so it may not have completion makers created. 

from the reconcile standpoint, we need to get hold of all Inprogress files. does not matter whether they are completed or not.

Reply:
It's obstacle to see the reason, let's add sone document about it.